src/CommunityToolkit.Maui/Behaviors/ICommunityToolkitBehavior.shared.cs

```diff
@@ -26,14 +26,9 @@ internal bool TrySetBindingContextToAttachedViewBindingContext()
 			return false;
 		}
 
-		behavior.SetBinding(BindableObject.BindingContextProperty, new Binding
-		{
-			Source = View,
-			Path = BindableObject.BindingContextProperty.PropertyName
-		});
+		behavior.SetBinding(BindableObject.BindingContextProperty, BindingBase.Create(static (TView? x) => x?.BindingContext, source: View));
```

Comment:
I tried this locally but I see the following error:

> Error BSG0007 : The lambda parameter type has to be declared as public, internal or protected internal.

Is it compiling for you?

Reply:
Weird. No, I'm not getting an error on this.

Reply:
Apologies - after `dotnet workload update` I'm seeing the same error

Reply:
Looks like it's a new Analyzer in .NET MAUI: https://github.com/dotnet/maui/blob/794a3895363f01ff40a1ee7dd980ab00fc55346a/src/Controls/src/BindingSourceGen/DiagnosticsFactory.cs#L85-L94

Since `BindableObject.BindingContext` is public, I can only assume it means that the source, `protected TView? View { get; set; }` must be public, internal or protected internal. However, when I modify the accessor on `View` to `protected internal`, I get the same error.

Reply:
The issue appears to be reporting that the type `TView` is not accessible based on this check:

<img width="559" alt="Screenshot 2024-09-23 at 21 09 37" src="https://github.com/user-attachments/assets/48e33387-c6b3-4cf7-945d-db7576a2a149">

Given @jkurdek authored this change I am hopeful you might be able to provide some insight into how we might fix this issue please?

We are currently trying to bind the `BindingContextProperty` of our behaviors to the `BindingContext` of the `View` that a behavior is attached to. How can we achieve this with a typed binding when our type is a generic type `where TView : Element`?

Reply:
@bijington thanks for looking into using expression based bindings! Unfortunately, we currently do not support using generics as part of the getter expression. We are aware of this limitation and we will consider fixing it in the future. 

Under the hood, the new expression-based `SetBinding` creates an instance of `TypedBinding` which is then used to call `SetBinding(property, typedBinding)`. I would suggest manually creating the appropriate `TypedBinding` instance in this case.

Reply:
@jkurdek thank you for the guidance! I have pushed up a change which I believe works around the issue. Almost have it compiling to be able to test it :)